examples/flava/native/train.py

```diff
@@ -143,6 +146,25 @@ def create_model(self) -> torch.nn.Module:
 
         model = model.to(self.device)
         print0(f"after moving to cuda: {torch.cuda.memory_allocated()/1024**3:.3} GB")
+
+        if self.config.training.activation_checkpointing:
+            check_fn = lambda submodule: isinstance(submodule, TransformerEncoderLayer)
+            if self.config.training.activation_checkpointing_reentrant:
+                checkpoint_impl = CheckpointImpl.REENTRANT
+            else:
+                checkpoint_impl = CheckpointImpl.NO_REENTRANT
+
+            non_reentrant_wrapper = partial(
+                checkpoint_wrapper,
+                offload_to_cpu=False,
+                checkpoint_impl=checkpoint_impl,
+            )
+            apply_activation_checkpointing_wrapper(
```

Comment:
I don't like that this is the state that we're in, but can we but apply_ac_wrapper after FSDP wrap if using non-reentrant, and before FSDP wrap using reentrant? This is the way that both of them would work (the difference is not good).

Reply:
do we want to keep non-reentrant if the memory usage is greater than reentrant?

Reply:
Good point, let's just remove non-reentrant.